### PR TITLE
Feature/app 810 assemble icon settings page

### PR DESCRIPTION
**Pull request by nirbhayel**

### CODE REVIEW

File: modules/settings/assets/js/components/bottom-bar/index.js [hunk 1]
```diff
@@ -1,8 +1,33 @@
...
+	const { selectedMenu, iconDesign, iconPosition } = useSettings();
+	const { save } = useStorage();
+	const { success, error } = useToastNotification();
+
+	const saveSettings = () => {
+		if (
+			selectedMenu.parent === 'widget' &&
+			selectedMenu.child === 'iconSettings'
+		) {
+			try {
```
Comment (DinetsV) [inline modules/settings/assets/js/components/bottom-bar/index.js:16]: Use async/await with try-catch or then/catch

**Review by pkniazevych — CHANGES_REQUESTED** [review]
Please, check Vasyl's comments

File: modules/settings/assets/js/app.js [hunk 2]
```diff
@@ -8,16 +8,20 @@ import {
...
+	useSavedSettings,
+} from '@ea11y/hooks';
 import { usePluginSettingsContext } from './contexts/plugin-settings';
 import { Sidebar } from './layouts/sidebar';
 
 const App = () => {
 	const { isConnected } = usePluginSettingsContext();
 	const { notificationMessage, notificationType } = useNotificationSettings();
 	const { selectedMenu } = useSettings();
+	useSavedSettings();
```
[the remark this answers is not in the document]
Comment (nirbhayel) [reply modules/settings/assets/js/app.js:24]: yes, it loads up the saved settings for icon design

File: modules/settings/assets/js/components/index.js [hunk 3]
```diff
@@ -3,11 +3,11 @@ export { default as Notifications } from './notifications';
...
-export { AdminTopBar } from './admin-top-bar';
-export { BottomBar } from './bottom-bar';
 export { default as IconSize } from './icon-size';
 export { default as IconSelect } from './icon-select';
 export { default as ColorPicker } from './color-picker';
 export { default as AlignmentMatrixControl } from './alignment-matrix-control';
 export { default as PositionControl } from './position-control';
+export { MenuItems } from '../components/sidebar-menu/menu';
+export { default as AdminTopBar } from './admin-top-bar';
+export { default as BottomBar } from './bottom-bar';
```
[the remark this answers is not in the document]
Comment (nirbhayel) [reply modules/settings/assets/js/components/index.js:13]: It is based on the previous ones we used and to keep them the same as per the last discussion.

File: modules/settings/assets/js/components/bottom-bar/index.js (hunk 1 above)
Comment (nirbhayel) [reply modules/settings/assets/js/components/bottom-bar/index.js:17]: added

File: modules/settings/assets/js/helpers/accessibility-options.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { __ } from '@wordpress/i18n';
+import {
+	AccessibilityPersonIcon,
+	AccessibilityTextIcon,
+	AccessibilityEyeIcon,
+	AccessibilityControlsIcon,
+} from '../icons';
+
+const options = () => {
+	const optionStyle = {
```
[the remark this answers is not in the document]
Comment (nirbhayel) [reply modules/settings/assets/js/helpers/accessibility-options.js:10]: updated

File: modules/settings/assets/js/components/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (DinetsV) [reply modules/settings/assets/js/components/index.js:13]: Ok but by default IDE import without "default" and I'm not sure that we have any rules for this

[the remark this answers is not in the document]
Comment (DinetsV) [reply modules/settings/assets/js/components/index.js:13]: Sorry I missed that it's an export

File: modules/settings/assets/js/components/bottom-bar/index.js [hunk 5]
```diff
@@ -1,8 +1,35 @@
...
+				await save({
+					a11y_widget_icon_settings: {
+						style: iconDesign,
+						position: iconPosition,
+					},
+				}).then(() => {
+					success('Settings saved!');
+					setHasChanges(false);
+				});
```
Comment (DinetsV) [inline modules/settings/assets/js/components/bottom-bar/index.js:26]: You dont need to use then after await just use 
```suggestion
        await save({
					a11y_widget_icon_settings: {
						style: iconDesign,
						position: iconPosition,
					},
				});
	success('Settings saved!');
	setHasChanges(false);
```

File: modules/settings/assets/js/app.js (hunk 2 above)
Comment (DinetsV) [inline modules/settings/assets/js/app.js:24]: Move call before initialize constants 
```suggestion
        useSavedSettings();
        const { isConnected } = usePluginSettingsContext();
	const { notificationMessage, notificationType } = useNotificationSettings();
	const { selectedMenu } = useSettings();
```

**Review by DinetsV — APPROVED** [review]

**Review by bainternet — APPROVED** [review]

**Pull request merged**